### PR TITLE
Fix xml config support for hot restart features

### PR DESCRIPTION
**Pull request by mmedenjak**
Add XML config support for hot backup and fix config options for cluster data recovery policy.

### CODE REVIEW

**Comment by mmedenjak** [comment]
run-lab-run

File: hazelcast-spring/src/main/resources/hazelcast-spring-3.8.xsd [hunk 1]
```diff
@@ -2836,6 +2843,17 @@
                 </xs:documentation>
             </xs:annotation>
         </xs:attribute>
+        <xs:attribute name="cluster-data-recovery-policy" type="xs:string" default="FULL_RECOVERY_ONLY">
```
Comment (vbekiaris) [inline hazelcast-spring/src/main/resources/hazelcast-spring-3.8.xsd]: Shouldn't this attribute's value be restricted to only valid `xs:enumeration` values?
```
                <xs:attribute name="cluster-data-recovery-policy">
                    <xs:simpleType>
                        <xs:restriction base="non-space-string">
                            <xs:enumeration value="FULL_RECOVERY_ONLY"/>
                            <xs:enumeration value="PARTIAL_RECOVERY_MOST_RECENT"/>
                            <xs:enumeration value="PARTIAL_RECOVERY_MOST_COMPLETE"/>
                        </xs:restriction>
                    </xs:simpleType>
                </xs:attribute>
```

**Comment by mmedenjak** [comment]
@vbekiaris Done.

File: hazelcast-spring/src/main/resources/hazelcast-spring-3.8.xsd [hunk 2]
```diff
@@ -2860,6 +2867,24 @@
                 </xs:documentation>
             </xs:annotation>
         </xs:attribute>
+        <xs:attribute name="cluster-data-recovery-policy" default="FULL_RECOVERY_ONLY">
+            <xs:annotation>
+                <xs:documentation>
+                    Specifies the policy that will be respected during hot restart cluster start.
+                    Valid values are;
+                    - FULL_RECOVERY_ONLY
```
Comment (jerrinot) [inline hazelcast-spring/src/main/resources/hazelcast-spring-3.8.xsd]: explanation would be nice here.

Comment (mmedenjak) [inline hazelcast-spring/src/main/resources/hazelcast-spring-3.8.xsd]: Added.

File: hazelcast-spring/src/main/resources/hazelcast-spring-3.8.xsd [hunk 3]
```diff
@@ -2835,6 +2835,13 @@
                     </xs:documentation>
                 </xs:annotation>
             </xs:element>
+            <xs:element name="backup-dir" type="xs:string" minOccurs="0" maxOccurs="1">
+                <xs:annotation>
+                    <xs:documentation>
+                        Base directory for hot backups. Each new backup will be created in a separate directory inside this one.
```
Comment (mmedenjak) [inline hazelcast-spring/src/main/resources/hazelcast-spring-3.8.xsd]: Added info, for hot restart dir too.

**Pull request merged**